### DISCUSSION TITLE
Can't install any language models

### DISCUSSION THREAD

**Discussion by FrankVanDorp** [explosion/spaCy#13162] [Help: Installation] [answered]
I want have installed spacy and I want to install en_core_web_sm-3.7.1, but nothing works.

I tried: python -m spacy download en_core_web_sm
But that threw the error: SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),)

I read this could be because of proxy issues, so I tried manually downloading the en_core_web_sm-3.7.1.tar.gz file and then use "pip install path/to/location/en_core_web_sm-3.7.1.tar.gz" as the readme instructed.
But I got the error: ERROR: Directory 'path/to/location' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.

How do I fix this?

**Comment by wesslen** [comment, marked as answer]
The SSLError you encountered when trying to download `en_core_web_sm` with `spacy download` indicates a problem with the SSL certificate verification while trying to access the package over HTTPS. This issue may indeed relate to your network's security setup, such as the presence of a proxy or firewall.

Since you've downloaded the `en_core_web_sm-3.7.1.tar.gz` file manually, you should be able to install it directly with pip, provided you supply the correct path to the file. The error message you're seeing suggests that the command might have been pointed to a directory instead of the tar.gz file, or the path was not specified correctly.

Here's how you can install the spaCy model `en_core_web_sm-3.7.1.tar.gz` file using pip:

1. Navigate to the directory where you've downloaded `en_core_web_sm-3.7.1.tar.gz`.
2. Run the following command, replacing `path/to/en_core_web_sm-3.7.1.tar.gz` with the actual path to the `tar.gz` file you've downloaded:
```
pip install /path/to/en_core_web_sm-3.7.1.tar.gz
```

Make sure to include the full file name, including the extension `.tar.gz`. Here’s an example if the file were in your current directory:
```
pip install ./en_core_web_sm-3.7.1.tar.gz
```

Give this a try and let me know if it resolves your issue. If you're still facing trouble, please ensure that the file path is correct and that the file is not corrupted. You can also try to install the package in a Python virtual environment to avoid potential conflicts with other installed packages.

**Reply by FrankVanDorp** [reply]
Thank you, ultimately this worked.
After retrying a few times, it turned out there was a space in one of the directory names and this is what caused the error when trying to install it manually.
So I moved it all to a different directory, and after updating python, conda and then spacy, I finally could install the en_core_web_sm package.
